README.md

```diff
@@ -1,131 +1,2469 @@
+ - [WritableDeviceRedundancyGroupRequest](docs/WritableDeviceRedundancyGroupRequest.md)
+ - [WritableDeviceRequest](docs/WritableDeviceRequest.md)
+ - [WritableDeviceTypeRequest](docs/WritableDeviceTypeRequest.md)
+ - [WritableFrontPortRequest](docs/WritableFrontPortRequest.md)
+ - [WritableFrontPortTemplateRequest](docs/WritableFrontPortTemplateRequest.md)
+ - [WritableInterfaceRedundancyGroupRequest](docs/WritableInterfaceRedundancyGroupRequest.md)
+ - [WritableInterfaceRequest](docs/WritableInterfaceRequest.md)
+ - [WritableInterfaceTemplateRequest](docs/WritableInterfaceTemplateRequest.md)
+ - [WritablePowerFeedRequest](docs/WritablePowerFeedRequest.md)
+ - [WritablePowerOutletRequest](docs/WritablePowerOutletRequest.md)
+ - [WritablePowerOutletTemplateRequest](docs/WritablePowerOutletTemplateRequest.md)
+ - [WritablePowerPortRequest](docs/WritablePowerPortRequest.md)
+ - [WritablePowerPortTemplateRequest](docs/WritablePowerPortTemplateRequest.md)
+ - [WritablePrefixRequest](docs/WritablePrefixRequest.md)
+ - [WritableRackRequest](docs/WritableRackRequest.md)
+ - [WritableRearPortRequest](docs/WritableRearPortRequest.md)
+ - [WritableRearPortTemplateRequest](docs/WritableRearPortTemplateRequest.md)
+ - [WritableServiceRequest](docs/WritableServiceRequest.md)
+ - [WritableVMInterfaceRequest](docs/WritableVMInterfaceRequest.md)
 
-```go
-package main
 
-import (
-	"context"
-	"fmt"
+## Documentation For Authorization
 
-	nb "github.com/nautobot/go-nautobot/pkg/nautobot"
-)
 
-func check(err error) {
-	if err != nil {
-		panic(err)
-	}
-}
+Authentication schemes defined for the API:
+### basicAuth
 
-func main() {
-	token, err := nb.NewSecurityProviderNautobotToken("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
-	check(err)
+- **Type**: HTTP basic authentication
 
-	c, err := nb.NewClientWithResponses(
-		"https://demo.nautobot.com/api/",
-		nb.WithRequestEditorFn(token.Intercept),
-	)
-	check(err)
+Example
 
-	ctx := context.Background()
+```go
+auth := context.WithValue(context.Background(), nautobot.ContextBasicAuth, nautobot.BasicAuth{
+	UserName: "username",
+	Password: "password",
+})
+r, err := client.Service.Operation(auth, args)
+```
 
-	resp, err := c.DcimManufacturersListWithResponse(ctx, &nb.DcimManufacturersListParams{})
-	check(err)
+### cookieAuth
 
-	fmt.Printf("%v", string(resp.Body))
+- **Type**: API key
+- **API key parameter name**: sessionid
+- **Location**: 
 
-}
+Note, each API key must be added to a map of `map[string]APIKey` where the key is: sessionid and passed in as the auth context for each request.
 
-```
+Example
 
-### Other usage examples
+```go
+auth := context.WithValue(
+		context.Background(),
+		nautobot.ContextAPIKeys,
+		map[string]nautobot.APIKey{
+			"sessionid": {Key: "API_KEY_STRING"},
+		},
+	)
+r, err := client.Service.Operation(auth, args)
+```
 
-- [Network Automation with Go Book](https://github.com/PacktPublishing/Network-Automation-with-Go/blob/main/ch06/nautobot/main.go)
```

Comment:
I think that having an example is great to get started, and adding the reference to this book it's good too even though is using an old version

Reply:
I changed the reference to the terraform provider, as that will use the new go-nautobot code :)